### PR TITLE
Feature/dev 4215f

### PR DESCRIPTION
**Pull request by Atash3000**

### CODE REVIEW

**Comment by adamdoe** [comment]
@Atash3000 can you assign a milestone for this PR?

File: packages/chart/src/components/EditorPanel/EditorPanel.tsx [hunk 1]
```diff
@@ -1976,7 +1976,7 @@ const EditorPanel = () => {
                         }
                         updateField={updateField}
                       />
-                      {false && visHasBrushChart && (
+                      {visHasBrushChart() && (
```
Comment (adamdoe) [inline packages/chart/src/components/EditorPanel/EditorPanel.tsx:1979]: Good!

File: packages/chart/src/components/EditorPanel/useEditorPermissions.js [hunk 2]
```diff
@@ -104,6 +104,7 @@ export const useEditorPermissions = () => {
     }
   }
   const visHasBrushChart = () => {
+    if (config.xAxis.type === 'categorical') return false
```
Comment (adamdoe) [inline packages/chart/src/components/EditorPanel/useEditorPermissions.js:107]: Good!

**Review by adamdoe — CHANGES_REQUESTED** [review]
Hey @atash3000, I didn't test this through the editor yet. Can you go through the first iteration of feedback and re-request a review when finished? Thanks!

File: packages/chart/src/components/BrushChart.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
+import { Group } from '@visx/group'
+import { useContext, useEffect, useRef, useState } from 'react'
+import ConfigContext from '../ConfigContext'
+import * as d3 from 'd3'
+import { scaleTime } from '@visx/scale'
+import { Text } from '@visx/text'
+
+function BrushChart({ xMax, yMax, xScaleBrush: x }) {
```
[the remark this answers is not in the document]
Comment (Atash3000) [reply packages/chart/src/components/BrushChart.tsx:8]: Resolved.

File: packages/chart/src/components/BrushChart.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  const { tableData, config, parseDate, setBrushConfig, getTextWidth } = useContext(ConfigContext)
+  const [brushState, setBrushState] = useState({ isBrushing: false, selection: [] })
+  const [tooltip, showTooltip] = useState(false)
+  const svgRef = useRef()
+  const brushheight = 25
+  const borderRadius = 15
+
+  const tooltipText = 'Drag edges to focus on a specific segment '
+  const textWidth = getTextWidth(tooltipText, `normal ${16 / 1.1}px sans-serif`)
```
[the remark this answers is not in the document]
Comment (Atash3000) [reply packages/chart/src/components/BrushChart.tsx:17]: fixed

File: packages/chart/src/components/BrushChart.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+  const calculateGroupTop = (): number => {
+    return Number(yMax) + config.xAxis.axisBBox + brushheight * 1.5
```
[the remark this answers is not in the document]
Comment (Atash3000) [reply packages/chart/src/components/BrushChart.tsx:20]: fixed

File: packages/chart/src/components/BrushChart.tsx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+  const handleMouseLeave = () => {
+    // hide tooltip text if brush was triggered
+    if (brushState.selection[0] !== 0 || brushState.selection[1] !== xMax) {
+      showTooltip(false)
+    }
+    showTooltip(false)
+  }
+
+  const brushHandle = (g, selection, firstDate, lastDate) => {
+    const textWidth = getTextWidth(firstDate, `normal ${16 / 1.1}px sans-serif`)
```
[the remark this answers is not in the document]
Comment (Atash3000) [reply packages/chart/src/components/BrushChart.tsx:38]: fixed

File: packages/chart/src/components/BrushChart.tsx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    return g
+      .selectAll('.handle--custom')
+      .data([{ side: 'left' }, { side: 'right' }])
+      .join(enter => {
+        const handleGroup = enter.append('g').attr('class', 'handle--custom')
+        // Define the path data for left and right handles
+        const leftHandlePath = 'M0.5,10A6,6 0 0 1 6.5,16V14A6,6 0 0 1 0.5,20ZM2.5,18V12M4.5,18V12'
+        const rightHandlePath = 'M6.5,10A6,6 0 0 0 0.5,16V14A6,6 0 0 0 6.5,20ZM4.5,18V12M2.5,18V12'
```
[the remark this answers is not in the document]
Comment (Atash3000) [reply packages/chart/src/components/BrushChart.tsx:46]: fixed

File: packages/chart/src/components/BrushChart.tsx [hunk 8]
```diff
@@ -0,0 +1,157 @@
+import { Group } from '@visx/group'
+import { useContext, useEffect, useRef, useState } from 'react'
+import ConfigContext from '../ConfigContext'
+import * as d3 from 'd3'
+import { scaleTime } from '@visx/scale'
+import { Text } from '@visx/text'
+
+function BrushChart({ xMax, yMax, xScaleBrush: x }) {
+  const { tableData, config, parseDate, setBrushConfig, getTextWidth } = useContext(ConfigContext)
```
[the remark this answers is not in the document]
Comment (Atash3000) [reply packages/chart/src/components/BrushChart.tsx:9]: fixed

File: packages/chart/src/components/BrushChart.tsx [hunk 9]
```diff
@@ -0,0 +1,157 @@
+import { Group } from '@visx/group'
+import { useContext, useEffect, useRef, useState } from 'react'
+import ConfigContext from '../ConfigContext'
+import * as d3 from 'd3'
+import { scaleTime } from '@visx/scale'
```
[the remark this answers is not in the document]
Comment (Atash3000) [reply packages/chart/src/components/BrushChart.tsx:5]: fixed

File: packages/chart/src/components/BrushChart.tsx [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+      .append('rect') // prettier-ignore
+      .attr('fill', '#949494')
+      .attr('stroke', '#c5c5c5')
+      .attr('stroke-width', 2)
+      .attr('ry', borderRadius)
+      .attr('rx', borderRadius)
+      .attr('height', brushheight)
+      .attr('width', xMax)
+
+    const safeInvert = (xScale, value) => {
```
[the remark this answers is not in the document]
Comment (Atash3000) [reply packages/chart/src/components/BrushChart.tsx:86]: Fixed

**Pull request merged**